### DISCUSSION TITLE
Chatgpt 4

### DISCUSSION THREAD

**Discussion by samfricher** [pashpashpash/vault-ai#37] [Q&A]
Is it possible to configure this with chatgpt 4? I have paid for chatgpt 4. If it is possible, how do I do this?

**Comment by pashpashpash** [comment]
The OpenAI plus $20/month subscription is a different product from the OpenAI API. OpenAI api access to the gpt4 model is in limited access right now, and you have to apply to get it. If you have access to that, you can use your API key and update the model used in the code here: https://github.com/pashpashpash/vault-ai/blob/master/vault-web-server/postapi/questions.go#L81-L84

By default, the Vault uses GPT 3.5 turbo to answer questions.

**Reply by tenuss2012** [reply]
What model name do you use?

**Reply by Bergdoktor** [reply]
Quoting directly from the source code of the abovementioned questions.go:

`model := openai.GPT3Dot5Turbo
	if form.Model == "GPT Davinci" {
		model = openai.GPT3TextDavinci003
	}
`

**Reply by tenuss2012** [reply]
I meant how to integrate gpt-4 model into it, for those with gpt-4 API access.

**Comment by Senethys** [comment]
> I meant how to integrate gpt-4 model into it, for those with gpt-4 API access.

You can´t. Being a paid subscriber is not the same as having access to the API.
